python/gui/auto_generated/qgsfilewidget.sip.in

```diff
@@ -75,6 +77,104 @@ Sets the current file ``path``.
     void setReadOnly( bool readOnly );
 %Docstring
 Sets whether the widget should be read only.
+%End
+
+    void setStorageType( const QString &storageType );
```

Comment:
Instead of adding all this to QgsFileWidget, have you considered subclassing QgsFileWidget with to make a more specialised widget? That would help us leave the widely-used QgsFileWidget minimal...

Reply:
I changed my mind many times about this. It's maybe indeed better separate both widget. Done c9cd7d5